lib/puppet-syntax/manifests.rb

```diff
@@ -15,7 +15,7 @@ def check(filelist)
       end
 
       # Catch syntax warnings.
-      Puppet::Util::Log.newdestination(Puppet::Test::LogCollector.new(errors))
+      Puppet::Util::Log.newdestination(Puppet::Test::LogCollector.new(output))
       Puppet::Util::Log.level = :warning
```

Comment:
I called it warnings since that's what `puppet parser validate` calls them when run on the command line:

```shell
$ puppet parser validate deprecation_notice.pp
Warning: The use of 'import' is deprecated at /home/phil/.bundler/ruby/1.9.1/puppet-syntax-120c16771034/spec/fixtures/test_module/manifests/deprecation_notice.pp:3. See http://links.puppetlabs.com/puppet-import-deprecation
   (at /var/lib/gems/1.9.1/gems/puppet-3.7.3/lib/puppet/parser/parser_support.rb:110:in `import')
Warning: Deprecation notice: Node inheritance is not supported in Puppet >= 4.0.0. See http://links.puppetlabs.com/puppet-node-inheritance-deprecation
   (at /home/phil/.bundler/ruby/1.9.1/puppet-syntax-120c16771034/spec/fixtures/test_module/manifests/deprecation_notice.pp:1)
```